### PR TITLE
Add repo api support for vstest repo.

### LINKED ISSUE #178: Remove patches from vstest
Remove all from https://github.com/dotnet/source-build/tree/dev/release/2.0/patches/vstest

### PR DESCRIPTION
**Pull request by codito**
* Remove patches for vstest repo
* Remove dependency on mono for desktop builds

Fix #178, #96

### CODE REVIEW

**Comment by eerhardt** [comment]
The errors I'm seeing:

```
  dotnet-install: Downloading link: https://dotnetcli.azureedge.net/dotnet/Runtime/1.0.5/dotnet-runtime-1.0.5-linux-x64.tar.gz
curl : (22) The requested URL returned error : 404 Not Found [/mnt/j/workspace/dotnet_source-build/dev_release_2.0/Ubuntu16.04_Release_prtest/repos/vstest.proj]
  dotnet-install: Cannot download: https://dotnetcli.azureedge.net/dotnet/Runtime/1.0.5/dotnet-runtime-1.0.5-linux-x64.tar.gz
  dotnet-install: Downloading legacy link: https://dotnetcli.azureedge.net/dotnet/Runtime/1.0.5/dotnet-ubuntu.16.04-x64.1.0.5.tar.gz
  
  dotnet-install: Downloading link: https://dotnetcli.azureedge.net/dotnet/Runtime/1.1.2/dotnet-runtime-1.1.2-linux-x64.tar.gz
curl : (22) The requested URL returned error : 404 Not Found [/mnt/j/workspace/dotnet_source-build/dev_release_2.0/Ubuntu16.04_Release_prtest/repos/vstest.proj]
```

Looks to me that something is wrong.

When you are building in source build, you shouldn't be downloading extra .NET Core SDKs.

File: repos/vstest.proj [hunk 1]
```diff
@@ -4,7 +4,7 @@
   <PropertyGroup>
     <BuildCommand>$(ProjectDirectory)/build$(ShellExtension) -c $(Configuration) -r $(TargetRid) -v 15.3.0 -vs preview-20170628-02 -nomono</BuildCommand>
```
Comment (eerhardt) [inline repos/vstest.proj]: The `-nomono` option can now be removed.

**Comment by codito** [comment]
@eerhardt we're explicitly passing the `DotNetCliToolsDir` to vstest build script now, it should use the existing cli toolset.

**Comment by codito** [comment]
@dotnet-bot test Fedora24 Release please

**Comment by eerhardt** [comment]
test CentOS7.1 Release

**Pull request merged**